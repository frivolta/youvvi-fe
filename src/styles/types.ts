export interface Theme {
    primaryFont: string;
    colors: {
      primary: string;
      primaryHover: string;
      primaryActive: string;
      text: string;
      lightText: string;
      componentBackground: string;
      componentHover: string;
      needs: string;
      needsBackground: string;
      wants: string;
      wantsBackground: string;
      incomes: string;
      incomesBackground: string;
      error: string;
      errorBackground: string;
      background: string;
      border: string;
      placeholder: string;
    };
    fonts: {
      header: {
        size: string;
        weight: string;
      };
      default: {
        size: string;
        weight: string;
        secondaryWeight: string;
        tertiaryWeight: string;
      };
      small: {
        size: string;
        weight: string;
      };
    };
    space: {
      s: string;
      m: string;
      l: string;
      xl: string;
      xxl: string;
      max: string;
    };
    borderRadius: {
      base: string;
      large: string;
      circle: string;
    };
    shadows: {
      default: string;
      small: string;
    };
  }