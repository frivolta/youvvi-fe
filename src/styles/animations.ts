import { keyframes } from "styled-components";

export const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

export const scale = keyframes`
  0% {
    transform: scale(1)
  }
  100% {
    transform: scale(1.2)
  }
`;
