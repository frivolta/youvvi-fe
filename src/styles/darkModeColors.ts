export const risinBlack = {
    800: "#181A29", // Backgrounds and dark areas
  };
  
  export const spaceCadet = {
    800: "#202336", // 800: Background over risin Black (eg. Card)
    700: "#292E5F", // 700: Hoverable elemnts (eg. Card hover)
    500: "#323753", // 500: Borders
  };
  
  export const darkSlateBlue = {
    500: "#3A4294", // 500: Background for neonBlue (eg. Needs tag bg)
  };
  
  export const neonBlue = {
    400: "#707CFF", // 400: Hover for neonBlue (eg. Button)
    500: "#5C6AFF", // 500: Primary color
    700: "#3344FF", // 700: Active
    rgba: "rgba(92,106,255,0.3)",
  };
  
  export const romanSilver = {
    500: "#89889B", //500: Default text color
  };
  
  export const darkThemeNeutral = {
    100: "#FFFFFF", //100: Title and important colors
  };
  
  export const winterSky = {
    500: "#FF2773", //500: Secondary color (Wants)
    rgba: "rgba(255,39,115,0.3)",
  };
  
  export const emerald = {
    500: "#58D17E", //500: Tertiary color (Incomes / Savings)
    rgba: "rgba(88,209,126,0.3)",
  };
  
  export const vermillion = {
    500: "#E70000", //500: Tertiary color (Incomes / Savings)
    rgba: "rgba(231,0,0,0.3)",
  };