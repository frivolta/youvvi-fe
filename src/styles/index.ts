export * from "./typography";
export * from "./themes";
export * from "./Global";
export * from "./devices";
export * from "./spaces";