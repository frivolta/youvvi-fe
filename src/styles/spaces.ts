export const spaceUnit = 16;
export const headerHeight = 72;
export const footerHeight = 55;
