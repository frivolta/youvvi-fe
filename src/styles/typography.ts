import styled from "styled-components";
export const primaryFont = '"Poppins", sans-serif';
export type TypographyProps = {
  color?: string;
  size?: string;
  weight?: string;
  margin?: string;
};

export const typeScale = {
  header1: "1.8rem",
  header2: "1.6rem",
  header3: "1.4rem",
  header4: "1.2rem",
  header5: "1.1rem",
  paragraph: "1rem",
  helperText: "0.8rem",
  copyrightText: "0.7rem",
  smallText: "0.7rem"
};

// Define typography styles
export const H1 = styled.h1<TypographyProps>`
  font-size: ${typeScale.header1};
  line-height: 42px;
  font-weight: 700;
  color: ${(props) =>
    props.color ? props.color : props.theme.colors.lightText};
`;

export const H2 = styled.h2<TypographyProps>`
  font-size: ${typeScale.header2};
  font-weight: ${(props) => (props.weight ? props.weight : "bold")};
  color: ${(props) =>
    props.color ? props.color : props.theme.colors.lightText};
`;

export const H3 = styled.h3<TypographyProps>`
  font-size: ${typeScale.header3};
  font-weight: ${(props) => (props.weight ? props.weight : "bold")};
  color: ${(props) =>
  props.color ? props.color : props.theme.colors.lightText};
`;

export const H4 = styled.h4<TypographyProps>`
  font-size: ${typeScale.header4};
  font-weight: ${(props) => (props.weight ? props.weight : "bold")};
  color: ${(props) =>
    props.color ? props.color : props.theme.colors.lightText};
`;

export const H5 = styled.h5<TypographyProps>`
  font-size: ${typeScale.header5};
  font-weight: ${(props) => (props.weight ? props.weight : "bold")};
  color: ${(props) =>
    props.color ? props.color : props.theme.colors.lightText};
`;

export const H6 = styled.h6<TypographyProps>`
  font-size: ${typeScale.helperText};
  font-weight: ${(props) => (props.weight ? props.weight : "bold")};
  color: ${(props) =>
    props.color ? props.color : props.theme.colors.lightText};
`;

export const Text = styled.p`
  font-size: ${typeScale.paragraph};
  color: ${(props) => (props.color ? props.color : props.theme.colors.text)};
`;

export const Description = styled.span`
  font-size: ${typeScale.helperText};
  color: ${(props) => (props.color ? props.color : props.theme.colors.text)};
`;

export const Amount = styled.span`
  font-size: ${typeScale.paragraph};
  color: ${(props) => (props.color ? props.color : props.theme.colors.text)};
`;

export const Span = styled.span<TypographyProps>`
  color: ${(props) => (props.color ? props.color : props.theme.colors.text)};
`;

export const Tag = styled.span<TypographyProps>`
  color: ${(props) => (props.color ? props.color : props.theme.colors.text)};
  font-size: ${typeScale.helperText};
`;

export const ExpenseTag = styled.span<TypographyProps>`
  color: ${(props) => (props.color ? props.color : props.theme.colors.text)};
  font-size: ${typeScale.smallText};
  font-weight: 700;
`;

export const Small = styled.small`
  font-size: ${typeScale.copyrightText};
`;