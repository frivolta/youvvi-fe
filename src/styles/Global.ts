import { primaryFont } from "./typography";
import { normalize } from "polished";
import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
${normalize()}
@import url('https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,300;0,400;0,500;0,600;0,700;0,800;1,300;1,400;1,500;1,600;1,700&display=swap');
html {
  box-sizing: border-box;
  font-size: 16px;
  min-height:100%;
    height: 100vh;
}
*, *:before, *:after {
  box-sizing: inherit;
}
body {
  margin: 0;
  font-family: ${primaryFont};
  line-height: 22px;
  min-height:100%;
  height: 100vh;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
main {
  width: 90%;
  margin: 0 auto;
}
#root{
  min-height:100%;
    height: 100vh;
}
`;