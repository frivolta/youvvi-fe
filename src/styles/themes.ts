import {
    risinBlack,
    spaceCadet,
    neonBlue,
    romanSilver,
    darkThemeNeutral,
    winterSky,
    emerald,
    vermillion,
  } from "./darkModeColors";
  import { primaryFont, typeScale } from "./typography";
  import { Theme } from "./types";
  
  export const defaultTheme: Theme = {
    primaryFont: primaryFont,
    colors: {
      primary: neonBlue[500],
      primaryHover: neonBlue[400],
      primaryActive: neonBlue[700],
      text: romanSilver[500],
      lightText: darkThemeNeutral[100],
      componentBackground: spaceCadet[800],
      componentHover: spaceCadet[700],
      needs: neonBlue[500],
      needsBackground: neonBlue.rgba,
      wants: winterSky[500],
      wantsBackground: winterSky.rgba,
      incomes: emerald[500],
      incomesBackground: emerald.rgba,
      error: vermillion[500],
      errorBackground: vermillion.rgba,
      background: risinBlack[800],
      border: spaceCadet[500],
      placeholder: romanSilver[500],
    },
    fonts: {
      header: {
        size: typeScale.paragraph,
        weight: "600",
      },
      default: {
        size: typeScale.paragraph,
        weight: "500",
        secondaryWeight: "300",
        tertiaryWeight: "700",
      },
      small: {
        size: typeScale.helperText,
        weight: "500",
      },
    },
    space: {
      s: "0.25rem" /* 4px */,
      m: "0.5rem" /* 8px */,
      l: "0.75rem" /* 12px */,
      xl: "1rem" /* 16px */,
      xxl: "1.5rem" /* 24px */,
      max: "2rem" /* 32px */,
    },
    borderRadius: {
      base: "5px",
      large: "15px",
      circle: "50%",
    },
    shadows: {
      default: "2px 0px 15px rgba(0, 0, 0, 0.05)",
      small: "0px 2px 10px rgba(0, 0, 0, 0.25)",
    },
  };