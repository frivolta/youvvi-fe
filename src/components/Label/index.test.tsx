import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Label } from "./";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../../../styles";

describe("<Label/>", () => {
  it("renders without errors with children", () => {
    const { getByTestId, getByText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Label>Children</Label>
      </ThemeProvider>
    );
    const element = getByTestId("Label");
    expect(element).toBeInTheDocument();
    expect(getByText("Children")).toBeInTheDocument();
  });

  it("renders with a different color", () => {
    const { getByText } = render(
      <ThemeProvider theme={defaultTheme}>
        <Label color="#000000">Children</Label>
      </ThemeProvider>
    );
    const element = getByText("Children");
    expect(element).toMatchSnapshot(
      `<span  color="#000000" data-testid="Label">Children</span>`
    );
  });
});
