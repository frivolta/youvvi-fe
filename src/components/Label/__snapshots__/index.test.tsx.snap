// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label/> renders with a different color: <span  color="#000000" data-testid="Label">Children</span> 1`] = `
<span
  class="sc-bdnylx eoVEFh"
  color="#000000"
  data-testid="Label"
>
  Children
</span>
`;
